test/staging/sm/Array/sort_small.js

```diff
@@ -2,23 +2,32 @@
 // This code is governed by the BSD license found in the LICENSE file.
 
 /*---
-includes: [sm/non262.js, sm/non262-shell.js]
-flags:
-  - noStrict
+includes: [compareArray.js]
 description: |
-  pending
+  Sort every possible permutation of some arrays.
 esid: pending
 ---*/
-// Sort every possible permutation of some arrays.
-function sortAllPermutations(data, comparefn) {
-    for (let permutation of Permutations(Array.from(data))) {
-        let sorted = (Array.from(permutation)).sort(comparefn);
-        for (let i in sorted) {
-            assert.sameValue(sorted[i], data[i],
-            [`[${permutation}].sort(${comparefn})`,
-            `returned ${sorted}, expected ${data}`].join(' '));
-        }
+
+// Yield every permutation of the elements in some array.
+function* Permutations(items) {
+  if (items.length == 0) {
+    yield [];
+  } else {
+    for (let i = 0; i < items.length; i++) {
+      let tail = items.slice(0);
+      let head = tail.splice(i, 1);
+      for (let e of Permutations(tail)) {
+        yield head.concat(e);
+      }
     }
+  }
+}
+
+function sortAllPermutations(data, comparefn) {
+  for (let permutation of Permutations(data)) {
+    let sorted = permutation.sort(comparefn);
+    assert.compareArray(sorted, data);
```

Comment:
`sorted` is in fact now the same array as `permutation`, better to just use `permutation` to avoid confusion